### ISSUE TITLE
[BUG] DeerRun S500 Bike Integration

### ISSUE THREAD

**Issue by ecology-rocks** [cagnulein/qdomyos-zwift#2932]
**Describe the bug**
Adding DeerRun (or SupeRun - same tech I think) Stationary Bike to the supported devices.

If I let it autoscan, it doesn't pick up that "PitPatS01" is what it wants.
If I force it to connect through the Settings -> Bike Settings, it will connect, but it doesn't register any of the movement.
There's another device that shows up called Smart41810266 which is definitely coming from the bike, but it won't let me connect to it at all there. 

**To Reproduce**

**Expected behavior**
I'd like for the app to be able to pick up the bike's movements. They do work through the (awful) PitPat app. 

**Screenshots**
![image](https://github.com/user-attachments/assets/ea7d8291-5e67-4e93-b170-667e93f72521)
(did you want a different screenshot out of the nRF app?)

**Smartphone (please complete the following information):**
 - Device: Google Pixel 8a
 - OS: Android
 - Version 15 - AP4A.241205.013

**Append a debug log**

[debug-Mon_Dec_23_09_29_15_2024.log](https://github.com/user-attachments/files/18230424/debug-Mon_Dec_23_09_29_15_2024.log)
[debug-Mon_Dec_23_09_29_45_2024.log](https://github.com/user-attachments/files/18230425/debug-Mon_Dec_23_09_29_45_2024.log)
[debug-Mon_Dec_23_09_28_58_2024.log](https://github.com/user-attachments/files/18230426/debug-Mon_Dec_23_09_28_58_2024.log)

(sorry idk if those are useful)
  
**Additional context**

**Comment by cagnulein** [comment]
Hi thanks. Do you have an app that already work with this bike?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 23 dic 2024 alle 15:37 Sam Lynn ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
perfect using that app follow this https://github.com/cagnulein/qdomyos-zwift/blob/master/docs/20_supported_devices_and_applications.md#android-device

**Comment by ecology-rocks** [comment]
[bugreport-akita-AP4A.241205.013-2024-12-25-16-20-04.zip](https://github.com/user-attachments/files/18247241/bugreport-akita-AP4A.241205.013-2024-12-25-16-20-04.zip)
 @cagnulein  Let's see if that captured it!

**Comment by cagnulein** [comment]
Perfect, I will check it ASAP!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 25 dic 2024 alle 22:24 Sam Lynn ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
got it @ecology-rocks how many resistance levels does it have?

**Comment by ecology-rocks** [comment]
@cagnulein 32!

**Comment by cagnulein** [comment]
just finished the first raw implementation

https://github.com/cagnulein/qdomyos-zwift/actions/runs/12504169774

Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

**Comment by ecology-rocks** [comment]
@cagnulein  I can't figure out how to download the build - there's nothing at the bottom of the page in either mobile or desktop browsers.

**Comment by ecology-rocks** [comment]
tried my best to do the bug report thing again with the bluetooth snooping. It won't let me upload (~26mb) -- https://drive.google.com/file/d/1f9n0G_Q7kJsK3izKhtHsI8dtyf_XYxFX/view?usp=sharing

**Comment by cagnulein** [comment]
Here https://github.com/cagnulein/qdomyos-zwift/actions/runs/12504169774/artifacts/2363407391

Simply create a debug log from qz and attach it here then https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Comment by ecology-rocks** [comment]
Got it - not sure which one, heres both. 

[debug-Tue_Dec_31_09_10_01_2024.log](https://github.com/user-attachments/files/18281991/debug-Tue_Dec_31_09_10_01_2024.log)
[debug-Tue_Dec_31_09_10_27_2024.log](https://github.com/user-attachments/files/18281992/debug-Tue_Dec_31_09_10_27_2024.log)

**Comment by cagnulein** [comment]
did qz crashed? because in the first log i see the bike but the log is truncated. if it doesn't crash, could you please redo it again?

**Comment by ecology-rocks** [comment]
Yes, if I have the bike on and I try to open the QZ app, it crashes immediately.

**Comment by cagnulein** [comment]
got it, there was a stupid error in the startup https://github.com/cagnulein/qdomyos-zwift/actions/runs/12571152090

we should need some versions to stabilize it, let me know

**Comment by ecology-rocks** [comment]
I think we're almost there! It connects. It doesn't read the resistance when I set it manually on the bike. If I try to set the resistance on the app, though, it resets the bike's resistance back down to 1. The resistance on the app won't increment past 1, it only goes 0-1. 
[debug-Wed_Jan_1_10_57_19_2025.log](https://github.com/user-attachments/files/18285901/debug-Wed_Jan_1_10_57_19_2025.log)

**Comment by cagnulein** [comment]
Perfect! I will check it in the next days!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 1 gen 2025 alle 17:02 Sam Lynn ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
i guess i fixed both https://github.com/cagnulein/qdomyos-zwift/actions/runs/12583417137

let me know!

**Comment by ecology-rocks** [comment]
It seems to work great now, thanks. Will the app lock up on me because it's one I downloaded from GitHub vs. the app store?

**Comment by cagnulein** [comment]
Yes exactly. I pushed the same version on the closed beta on android too. Send me your email to Roberto.viola83 at Gmail.com and I can add to the closed beta on the play store

**Comment by cagnulein** [comment]
@ecology-rocks are we fine with this?
